include/dynamatic/Dialect/Handshake/HandshakeArithOps.td

```diff
@@ -101,6 +101,43 @@ class Handshake_Arith_IToICastOp<string mnemonic, list<Trait> traits = []> :
   let hasVerifier = 1;
 }
 
+class Handshake_Arith_FToFCastOp<string mnemonic, list<Trait> traits = []> :
+  Handshake_Arith_Op<mnemonic, traits # [SameExtraSignalsInterface,
+                                         ReshapableChannelsInterface]
+> {
+  let arguments = (ins IntChannelType:$in);
+  let results = (outs IntChannelType:$out);
+  
+  let assemblyFormat = "$in attr-dict `:` type($in) `to` type($out)";
+
+  // TODO: add folding operation
+  let hasVerifier = 1;
+}
+
+class Handshake_Arith_IToFCastOp<string mnemonic, list<Trait> traits = []> :
+  Handshake_Arith_Op<mnemonic, traits # [ReshapableChannelsInterface]
+> {
+  let arguments = (ins IntChannelType:$in);
+  let results = (outs FloatChannelType:$out);
+  
+  let assemblyFormat = "$in attr-dict `:` type($in) `to` type($out)";
+
+  // TODO: add folding operation
+  let hasVerifier = 1;
+}
+
+class Handshake_Arith_FToICastOp<string mnemonic, list<Trait> traits = []> :
+  Handshake_Arith_Op<mnemonic, traits # [ReshapableChannelsInterface]
```

Comment:
Same as above, the `SameExtraSignalsInterface` should probably be present here as well.